modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -625,6 +664,9 @@ def _reset(df, *axis_lengths, partition_idx):
                     func=_reset,
                     enumerate_partitions=True,
                     new_columns=new_columns,
+                    dtypes=(
```

Comment:
this is to make the `test_merge_preserves_metadata` passing, otherwise, the result of the `.merge` would lose the dtypes cache during index drop at the end: https://github.com/modin-project/modin/blob/0f5e491cc5e2c458a4e227b1d24fc0a09a027793/modin/core/storage_formats/pandas/query_compiler.py#L556

I don't really think we need a separate PR for that as this change is really straight forward and is also related to `.merge` dtypes cache